site/public/js/forum.js

```diff
@@ -2103,3 +2103,63 @@ $(() => {
 function scrollThreadListTo(element){
   $(element).get(0).scrollIntoView({behavior: "smooth", block: "center"});
 }
+
+//Only used by the posters and only on recent posts (60 minutes since posted)
+function sendAnnouncement(thread_post_content, id){
+  $(".pin-and-email-message").attr('disabled','disabled');
+  $.ajax({
+    type: 'POST',
+    url: buildCourseUrl(['forum', 'make_announcement']),
+    data: {"id": id, 'csrf_token': window.csrfToken},
+    success: function(data){
+      if (JSON.parse(data).status === "success"){ 
```

Comment:
Good practice to wrap any `JSON.parse` with a try/catch since if the server returns a bad response (e.g. robot exception page), then this will just raise an exception without alerting the user anything has gone wrong.